### PR TITLE
Update README

### LINKED ISSUE #19: Update README
Better documentation for methods in condition_base needs to be done as part of this issue.

### PR DESCRIPTION
**Pull request by cameron1729**
Fixes #19

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,17 +1,40 @@
...
-TODO Provide more detailed description here.
+Rules are what determine if a user will be added or removed from a cohort. A rule is defined by two things:
 
-## Versions and branches ##
+1. A cohort
+2. A set of conditions
+
+For a user to be added to the cohort specified by a rule, they must match all of the rule's conditions. *NB:* A cohort can be managed by *one and only one* rule. This is to prevent rules competing over users in a cohort (e.g., to avoid situations where Rule A wants users a, b, c to be in a cohort, but Rule B wants to remove user c from the same cohort). In addition, only manually created cohorts can be managed by tool_cohortmanager.
+
+Rules are processed by cron. Depending on how many users are matched by a rule, this process can take some time. For rules matching large sets of users, some configuration options are provided which may be useful to server administrators.
```
Comment (dmitriim) [inline README.md:28]: Maybe it's worse to mention here that some rules can be triggered on specific events if related conditions require that. E.g. see https://github.com/catalyst/moodle-tool_cohortmanager/blob/MOODLE_39_STABLE/classes/tool_cohortmanager/condition/user_profile.php#L464

File: README.md [hunk 2]
```diff
@@ -35,6 +58,57 @@ Alternatively, you can run
...
+```
+tool_cohortmanager
+└── classes
+    └── tool_cohortmanager
+        └── condition
+            ├── user_custom_profile.php
+            └── user_profile.php
+```
+
+`user_custom_profile.php` and `user_profile.php` are the definitions for the User custom profile field and User standard profile field rules, respectively. For implementation details, see these files. Any plugin wishing to add a rule must simply add similar classes inside its own directory.
```
Comment (dmitriim) [inline README.md:104]: Maybe good to add that conditions must implement base_condition class

File: README.md [hunk 3]
```diff
@@ -26,15 +51,73 @@ The plugin can be also installed by putting the contents of this directory to
...
 
+# Configuration
+
+This section will cover how to configure a basic rule.
+
+## Prerequisites
+1. At least one manually created cohort (See _Site administration > Users > Cohorts_)
+
+## Creating a rule
+1. Navigate to _Site administration > Plugins > Admin tools > Cohort manager > Manage dynamic cohort rules_
```
Comment (dmitriim) [inline README.md:70]:
```suggestion
1. Navigate to _ Site administration -> Users -> Accounts -> Cohort manager -> Manage dynamic cohort rules_
```

Comment (dmitriim) [inline README.md:70]:
```suggestion
1. Navigate to _Site administration > Users -> Accounts > Cohort manager > Manage dynamic cohort rules_
```

**Comment by golenkovm** [comment]
Clear and sufficient instructions, easy to follow, great work @cameron1729 and @dmitriim

**Review by golenkovm — APPROVED** [review]

**Pull request merged**